src/plugins/analysis/init_systems/test/test_plugin_init_system.py

```diff
@@ -28,26 +28,24 @@ def setUpClass(self):
         }
 
         for test_file, path in test_files.items():
-            exec("self.test_file_" + test_file + " = FileObject(file_path=os.path.join(test_init_dir, path))")
-            exec("self.test_file_" + test_file + ".processed_analysis['file_type'] = {'mime': 'text/plain'}")
-            exec("self.test_file_" + test_file + ".root_uid = self.test_file_" + test_file + ".uid")
-            exec("self.test_file_" + test_file + ".virtual_file_path = {self.test_file_" + test_file + ".get_root_uid(): [\"" + path + "\"]}")
+            test_fo = FileObject(file_path=os.path.join(test_init_dir, path))
+            setattr(cls, 'test_file_{}'.format(test_file), test_fo)
+            test_fo.processed_analysis['file_type'] = {'mime': 'text/plain'}
+            test_fo.root_uid = test_fo.uid
+            test_fo.virtual_file_path = {test_fo.get_root_uid(): [path]}
 
-        self.test_file_not_text = FileObject(file_path="{}etc/systemd/system/foobar".format(test_init_dir))
-        self.test_file_not_text.processed_analysis['file_type'] = {'mime': 'application/zip'}
+        cls.test_file_not_text = FileObject(file_path="{}etc/systemd/system/foobar".format(test_init_dir))
+        cls.test_file_not_text.processed_analysis['file_type'] = {'mime': 'application/zip'}
 
     @classmethod
-    def tearDownClass(self):
-        super(TestAnalysisPluginInit, self).tearDownClass()
+    def tearDownClass(cls):
```

Comment:
Is that even necessary? I'd have guessed, the parents function is called anyway

Reply:
AFAIK this replaces the parent's function just like `tearDown` would (or any other function of a subclass), so you need to manually call `super().tearDownClass()`

Reply:
I think you're right. Removed.